msprime/trees.py

```diff
@@ -289,11 +289,17 @@ def __init__(
         self._tree = tree
         self._x_coords = {}
         self._y_coords = {}
+        try:
+            self._colour = {node:col for node,col in node_colours.keys()}
+        except:
```

Comment:
Blanket swallowing all exceptions is a very bad idea. Something catastrophic could have happened in this block which would be silently passed over.